src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -179,12 +201,8 @@ public void Initialize()
         // C# may optimize away the pinned local, producing incorrect results.
         static internal unsafe byte* GetAddrOfPinnedArrayFromEETypeField(IntPtr* ppEEType)
         {
-#if BIT64
-            // on 64-bit there is an extra 4 byte padding before the array data starts
-            return (byte*)ppEEType + sizeof(EETypePtr) + (2 * sizeof(int));
-#else
-            return (byte*)ppEEType + sizeof(EETypePtr) + sizeof(int);
-#endif
+            // on 64-bit there is an extra 4 byte padding (DIMENSIONS_DELTA) before the array data starts
+            return (byte*)ppEEType + sizeof(EETypePtr) + ((1 + DIMENSIONS_DELTA) * sizeof(int));
```

Comment:
The `DIMENSIONS_DELTA` name make sense for multi-dim arrays ... but it is used for single dim arrays here. It may be better to call it `PADDING` or `PADDING_DELTA`.